src/DashboardService.Business/Board/EditBoardCommand.cs

```diff
@@ -19,23 +25,50 @@ public class EditBoardCommand : IEditBoardCommand
   private readonly IHttpContextAccessor _httpContext;
   private readonly IAccessValidator _accessValidator;
   private readonly IResponseCreator _responseCreator;
+  private readonly IEditBoardRequestValidator _validator;
 
   public EditBoardCommand(
     IBoardRepository repository,
     IPatchDbBoardMapper _patchMapper,
     IHttpContextAccessor httpContextAccessor,
     IAccessValidator accessValidator,
-    IResponseCreator responseCreator)
+    IResponseCreator responseCreator,
+    IEditBoardRequestValidator validator)
   {
     _boardRepository = repository;
     _patchDbBoardMapper = _patchMapper;
     _httpContext = httpContextAccessor;
     _accessValidator = accessValidator;
     _responseCreator = responseCreator;
+    _validator = validator;
   }
 
-  public Task<OperationResultResponse<bool>> ExecuteAsync(Guid id, JsonPatchDocument<PatchBoardRequest> request)
+  public async Task<OperationResultResponse<bool>> ExecuteAsync(
+    Guid id,
+    JsonPatchDocument<PatchBoardRequest> request,
+    CancellationToken ct)
   {
-    throw new NotImplementedException();
+    if (!await _accessValidator.IsAdminAsync())
+    {
+      return _responseCreator.CreateFailureResponse<bool>(HttpStatusCode.Forbidden);
+    }
+
+    ValidationResult validationResult = await _validator.ValidateAsync(
+      (await _boardRepository.GetAsync(id, new(), ct), request),
+      ct);
+
+    if (!validationResult.IsValid)
+    {
+      return _responseCreator.CreateFailureResponse<bool>(
+        HttpStatusCode.BadRequest,
+        validationResult.Errors.ConvertAll(vf => vf.ErrorMessage));
+    }
+
+    return new OperationResultResponse<bool>(
+      body: await _boardRepository.EditByIdAsync(
+        id,
+        _httpContext.HttpContext.GetUserId(),
+        _patchDbBoardMapper.Map(request),
+        ct));
```

Comment:
this is impossible to read. Split into several steps

Reply:
++